common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -266,12 +271,37 @@ def destroy(self, request, uuid):
                 data='The Entitlement for this UUID does not exist or is Expired.'
             )
 
-        if entitlement.enrollment_course_run is None:
-            return Response(status=status.HTTP_204_NO_CONTENT)
+        with transaction.atomic():
+
+            if is_revoke and entitlement.is_entitlement_refundable():
+                if entitlement.enrollment_course_run is not None:
```

Comment:
Both branches of the above 'if' block do this check.  Why not pull it out above the 'if' block?

Reply:
What do you mean by "block the other logic within the code path"?

I'm just suggesting that instead of copy and pasting the "if entitlement.enrollment_course_run is not None" block of code at the beginning of both branches here, just pull it above and only have that block once.  DRY